run-tests.js

```diff
@@ -400,14 +400,12 @@ ${ENDGROUP}`)
 
   const hasIsolatedTests = tests.some((test) => {
     return configuredTestTypes.some(
-      (type) =>
-        type !== testFilters.unit && test.file.startsWith(`test/${type}`)
+      (type) => typeof type === 'string' && test.file.startsWith(type)
```

Comment:
This seems incorrect, shouldn't we be checking if any of the tests match `production`, `development`, or `e2e`? I don't think we correctly since some of these are RegExp's

Reply:
Those are the five configured test type values:

```js
[
  new RegExp('^(test/(development|e2e)|packages/.*/src/.*)/.*\\.test\\.(js|jsx|ts|tsx)$'),
  new RegExp('^(test/(production|e2e))/.*\\.test\\.(js|jsx|ts|tsx)$'),
  new RegExp('^test/unit|packages/.*/src/.*/.*\\.test\\.(js|jsx|ts|tsx)$'),
  'examples/',
  'test/integration/',
  'test/e2e/'
]
```

Before this change, we coerced the first two RegExps to a string that never matched. And the three actual strings were prepended with `test/` incorrectly. Now, we use the three string values as-is since the test file might start with those. So we were and are not matching `development` and `production`. Should we?

Side note: This only affects Windows, because in the GitHub Actions we're passing an explicit `--type`, so this logic is not used at all there.

Reply:
Yeah `development` and `production` needs to be matched for isolated case as well

Reply:
This made me realise that the variable name `hasIsolatedTests` and the comments around it were outdated, since we already create the shared install dir also for integration tests (just not on Windows yet). Adjusted it accordingly in https://github.com/vercel/next.js/pull/68166/commits/517b84851297b3f29d1f0218d0b0fab26aa89d24.